storage/src/vespa/storage/common/cluster_state_bundle.h

```diff
@@ -0,0 +1,27 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include <vespa/document/bucket/bucketspace.h>
+
+namespace storage {
+
+namespace lib { class ClusterState; }
+
+/**
+ * Class representing the baseline cluster state and the derived cluster
+ * state for each bucket space.
+ */
+class ClusterStateBundle
+{
+    using ClusterState = lib::ClusterState;
+    std::shared_ptr<const ClusterState> _baselineClusterState;
+public:
+    ClusterStateBundle(const ClusterState &baselineClusterState);
```

Comment:
I think it would be nice to also have a factory function such as `shared_ptr<ClusterStateBundle> ClusterStateBundle::ofBaselineState(state)` or similar. But this can be done when multiple bucket spaces are added. At the moment this constructor is unambiguous.